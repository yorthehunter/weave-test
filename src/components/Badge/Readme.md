Badge:

```js
<span>Hello everyone. <Badge>!</Badge></span>
```

Badge using `content` prop:

```js
<span>Hello everyone. <Badge content="1" /></span>
```

`Position: upperRight`

```js
<div style={{ outline: '1px solid red', position: 'relative', padding: '1rem' }}>
  Hello everyone.
  <Badge content="1" position="upper-right" />
</div>
```

`Position: inline`

```js
<span>Hello everyone. <Badge content="1" position="inline" /></span>
```
