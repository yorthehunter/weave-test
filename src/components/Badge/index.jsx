import React from 'react';
import { string, oneOf } from 'prop-types';
import 'weave-ui-badge';
import { positionModifier } from '../utilities/util';
import CustomTag from '../utilities/CustomTag';

const componentPrefix = 'weave-badge';

/**
 * A badge is a small, round indicator that often sits inline or slightly overlapping
 * content of which it relates. Badges are subtle ways to "notify" the user,
 * often to signify that action is required because of a change in status.
 */

const Badge = (props) => {
  const {
    position,
    content,
    children,
  } = props;

  return (
    <CustomTag componentPrefix={componentPrefix} {...props}>
      {content || children}
    </CustomTag>
  );
};

const defaultProps = {
  tag: 'span',
  position: null,
  content: '',
  children: false,
};

const propTypes = {
  tag: string,
  position: oneOf([
    'inline',
    'upper-right',
    null,
  ]),
  /** Badge content (if props.children aren't specified) */
  content: (props) => {
    if (props.content && props.children) {
      console.warn('Cannot have both "content" AND "children" props.');
    }
  },
  /** Badge content (if props.content isn't specified) */
  children: (props) => {
    if (props.content && props.children) {
      console.warn('Cannot have both "content" AND "children" props.');
    }
  },
};

Badge.defaultProps = defaultProps;
Badge.propTypes = propTypes;
export default Badge;
