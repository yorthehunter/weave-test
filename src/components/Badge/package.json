{
  "name": "weave-react-banner",
  "dependencies": {
    "weave-react-utilities": "file:../utilities",
    "weave-ui-badge": "file:../../weave/components/badge"
  },
  "version": "1.0.0",
  "description": "Weave banner React component",
  "main": "index.jsx",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/stitchfix/weave.git"
  },
  "keywords": [
    "css",
    "scss",
    "weave",
    "design",
    "system",
    "style",
    "guide",
    "component",
    "library",
    "ui"
  ],
  "author": "Stitch Fix",
  "license": "UNLICENSED",
  "bugs": {
    "url": "https://github.com/stitchfix/weave/issues"
  },
  "homepage": "https://github.com/stitchfix/weave#readme"
}
