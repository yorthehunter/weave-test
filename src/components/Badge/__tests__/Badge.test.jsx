import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Badge from '../';

describe('Badge', () => {
  it('renders correctly with default values', () => {
    const rendered = renderer.create(<Badge />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('warns in the console if both content and children exist', () => {
    // Removes console.warn from appearing the Jest coverage output
    jest.spyOn(console, 'warn').mockImplementation();

    expect(console.warn).not.toHaveBeenCalled();
    shallow(<Badge content="1">2</Badge>);
    expect(console.warn).toHaveBeenCalled();
  });
});
