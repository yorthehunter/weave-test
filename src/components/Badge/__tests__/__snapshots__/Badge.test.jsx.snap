// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly with default values 1`] = `
<span
  className="weave-badge"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;
